### DISCUSSION TITLE
When making a packaged WinUI3.0 app self contained is there a way to get rid of the dependency on Microsoft.VCLibs.140.00 & Microsoft.VCLibs.140.00.UWPDesktop packages

### DISCUSSION THREAD

**Discussion by koenvd** [microsoft/WindowsAppSDK#3524] [Q&A] [answered]
I succeeded to make a packaged WinUI3.0 app self contained so I no longer need to distribute the WindowsAppSDK appx package next to my app bundle.

However I still have a dependency on the CRT (as shown here)
<img width="681" alt="Screenshot 2023-03-21 at 17 50 13" src="https://user-images.githubusercontent.com/15209987/226683651-a8cb208e-9345-427a-9a90-e1232b4eba17.png">

But since the WASDK is built using the `HybridCrt` approach I would expect I no longer need to distribute these packages as well. Unless these are still required for the self contained .NET 6.0 parts that I'm using in the app.

So I'm just wondering if I could build my self contained app in such a way that both VCLibs packages are no longer outputted as a dependency of my app and my app becomes completely self contained?

**Comment by DarranRowe** [comment, marked as answer]
Digging into the Windows App SDK build files, there is an interesting section there.
```
  <!-- By default, add CRT package dependencies for UAP apps.  Previously, this was done to satisfy
      CRT linkage for WinUI 3 binaries.  With Hybrid CRT, that's no longer needed, but we retain this behavior 
      to avoid breaking existing projects.  At deployment time, it's very likely that a target device will already
      have Microsoft.VCLibs and Microsoft.VCLibs.Desktop FWPs installed.  But apps can still opt out of referencing
      these FWPs by setting WinUISDKReferences='false'. -->
  <PropertyGroup Condition="'$(WinUISDKReferences)'=='' and '$(TargetPlatformIdentifier)'=='UAP'">
    <WinUISDKReferences>true</WinUISDKReferences>
  </PropertyGroup>

  <ItemGroup Condition="'$(WinUISDKReferences)'=='true' and
                      ('$(OutputType)' == 'Exe' or 
                      '$(OutputType)' == 'AppContainerExe' or 
                      '$(MSBuildProjectExtension)'=='.wapproj' or
                      ('$(OutputType)'=='WinExe' and '$(EnableMsixTooling)'=='true'))">
    <SDKReference Include="Microsoft.VCLibs, Version=14.0" />
    <SDKReference Include="Microsoft.VCLibs.Desktop, Version=14.0" />
    <SDKReference Include="Microsoft.UniversalCRT.Debug, Version=$(TargetPlatformVersion)" Condition="'$(Configuration)' == 'Debug'" />
  </ItemGroup>
```
This is in the Microsoft.WinUI.AppX.targets file.
This means two things. First, this reference is coming from the Windows App SDK itself. Second, try placing `<WinUISDKReferences>false</WinUISDKReferences>` in the .wapproj file if it exists, or in the .csproj file if the .wapproj file doesn't exist. If you are placing it in the .wapproj file, look for the property group that contains the DefaultLanguage property. If you are putting it in the .csproj then placing it directly below UseWinUI is good enough. Finally, this requires the exact spelling and case.
As a note, since this also disables the debug UCRT reference for debug builds, then this can cause problems if your self contained application contains native components that are built to reference the debug UCRT.
---Edit---
I only just noticed that I forgot to mark the xml as code, so the tags were removed.

**Reply by koenvd** [reply]
Waw @DarranRowe . This indeed worked flawlessly. 

After adding `<WinUISDKReferences>false</WinUISDKReferences>` to the .wapproj my dependencies folder was gone 🥳.  (Not using native components as well at the moment).

Thanks for looking into this and providing this detailed answer. Really appreciated.

**Comment by Scottj1s** [comment]
Thanks for answering, @DarranRowe. Closing as this is now resolved.
